### PR TITLE
fixing the config for the most recent urdf

### PR DESCRIPTION
**Pull request by nlyubova**
There are many changes, mainly because of changes in URDF, So I re-generated the moveit_config with the Assistant.

### CODE REVIEW

File: config/NaoH25V40.srdf [hunk 1]
```diff
@@ -3,88 +3,112 @@
...
-	<chain base_link="torso" tip_link="l_sole" />
+        <chain base_link="torso" tip_link="l_sole" />
     </group>
     <group name="leg_right">
-	<chain base_link="torso" tip_link="r_sole" />
+        <chain base_link="torso" tip_link="r_sole" />
     </group>
-    <group name="hand_left">
-        <joint name="LWristYaw" />
+    <group name="left_hand">
```
Comment (mikaelarguedas) [inline config/NaoH25V40.srdf:27]: did you change the name of the group on purpose ? 
Can we keep the previous naming? which was: CHAINNAME_SIDE, e.g. leg_right, arm_right...

File: config/NaoH25V40.srdf [hunk 2]
```diff
@@ -3,88 +3,112 @@
...
-    </group>
     <group name="arm_hand_left">
-	<chain base_link="torso" tip_link="l_wrist" />
+        <chain base_link="torso" tip_link="l_wrist" />
     </group>
     <group name="arm_hand_right">
-	<chain base_link="torso" tip_link="r_wrist" />
+        <chain base_link="torso" tip_link="r_wrist" />
+    </group>
+    <group name="right_hand">
```
Comment (mikaelarguedas) [inline config/NaoH25V40.srdf:42]: same as previous comment on group naming

File: config/NaoH25V40.srdf [hunk 3]
```diff
@@ -3,88 +3,112 @@
...
-    <passive_joint name="RHipYawPitch" />
-    <passive_joint name="RFinger11" />
-    <passive_joint name="RFinger12" />
-    <passive_joint name="RFinger13" />
-    <passive_joint name="RFinger21" />
-    <passive_joint name="RFinger22" />
-    <passive_joint name="RFinger23" />
-    <passive_joint name="RThumb1" />
-    <passive_joint name="RThumb2" />
+    <passive_joint name="virtual_joint" />
```
Comment (mikaelarguedas) [inline config/NaoH25V40.srdf:89]: I think that the virtual joint is known to be virtual. I think you don't have to put it as passive joint.

File: config/NaoH25V40.srdf [hunk 4]
```diff
@@ -3,88 +3,112 @@
...
     <!--PASSIVE JOINT: Purpose: this element is used to mark joints that are not actuated-->
-    <passive_joint name="LFinger11" />
-    <passive_joint name="LFinger12" />
-    <passive_joint name="LFinger13" />
-    <passive_joint name="LFinger21" />
-    <passive_joint name="LFinger22" />
-    <passive_joint name="LFinger23" />
-    <passive_joint name="LThumb1" />
-    <passive_joint name="LThumb2" />
-    <passive_joint name="RHipYawPitch" />
```
Comment (mikaelarguedas) [inline config/NaoH25V40.srdf:62]: Same question as in DCM, what happens when you try to send commend to RHipYawPitch ?

File: config/controllers.yaml [hunk 5]
```diff
@@ -66,13 +68,13 @@ controller_list:
     type: FollowJointTrajectory
     default: true
     joints:
-      - LWristYaw
+#      - LWristYaw
```
Comment (mikaelarguedas) [inline config/controllers.yaml]: remove commented line (71 and 78)

File: config/fake_controllers.yaml [hunk 6]
```diff
@@ -1,46 +1,58 @@
 controller_list:
-  - name: fake_LeftArm_controller
+  - name: fake_left_arm_controller
```
Comment (mikaelarguedas) [inline config/fake_controllers.yaml]: keep naming conventions: CHAINNAME_SIDE, (e.g. leg_right, arm_right...)

File: launch/moveit_rviz.launch [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
   <arg name="config" default="false" />
   <arg unless="$(arg config)" name="command_args" value="" />
-  <arg     if="$(arg config)" name="command_args" value="-d $(find nao_moveit_config)/config/moveit.rviz" />
+  <arg     if="$(arg config)" name="command_args" value="-d $(find nao_moveit_config)/launch/moveit.rviz" />
```
Comment (mikaelarguedas) [inline launch/moveit_rviz.launch]: why do we have two urdf.rviz ? we don't use in config/moveit.rviz anymore ?

File: launch/planning_context.launch [hunk 8]
```diff
@@ -4,13 +4,13 @@
 
   <!-- The name of the parameter under which the URDF is loaded -->
   <arg name="robot_description" default="robot_description"/>
-  <arg name="version" default="V40"/>
 
   <!-- Load universal robot description format (URDF) -->
-  <param if="$(arg load_robot_description)" name="$(arg robot_description)" command="$(find xacro)/xacro.py '$(find nao_description)/urdf/nao$(arg version)_generated_urdf/nao_robot.xacro'"/>
-  <!-- The semantic description that corresponds to the URDF -->
-  <param name="$(arg robot_description)_semantic" textfile="$(find nao_moveit_config)/config/nao_robot_v4_H25.srdf" />
+  <param if="$(arg load_robot_description)" name="$(arg robot_description)" textfile="$(find nao_description)/urdf/naoV40_generated_urdf/nao.urdf"/>
```
Comment (mikaelarguedas) [inline launch/planning_context.launch]: it would be awesome to keep the robot description with xacro call and the version argument (as it was previously). This way moveit will keep working for all nao models and not only nao V40.

File: package.xml [hunk 9]
```diff
@@ -18,8 +18,6 @@
 
   <run_depend>nao_description</run_depend>
   <run_depend>joint_state_publisher</run_depend>
-  <run_depend>moveit_fake_controller_manager</run_depend>
-  <run_depend>moveit_simple_controller_manager</run_depend>
```
Comment (mikaelarguedas) [inline package.xml:22]: why did you remove the controller dependencies ? can you run demo.launch without those packages installed ?

File: config/NaoH25V40.srdf (hunk 1 above)
Comment (nlyubova) [inline config/NaoH25V40.srdf:27]: good question ;) yes I've changed it on purpose ! to make them compatible with the MoveIt simple grasps, their code needs the group names exactly like this "_hand". So, now they are named in the same way for all three robots.

File: config/NaoH25V40.srdf (hunk 2 above)
Comment (nlyubova) [inline config/NaoH25V40.srdf:42]: same answer, yes, definitely on purpose!

File: config/NaoH25V40.srdf (hunk 3 above)
Comment (nlyubova) [inline config/NaoH25V40.srdf:89]: right, I will add them back

File: config/controllers.yaml (hunk 5 above)
Comment (nlyubova) [inline config/controllers.yaml]: done

File: config/NaoH25V40.srdf (hunk 4 above)
Comment (nlyubova) [inline config/NaoH25V40.srdf:62]: ok, added it back

File: config/fake_controllers.yaml (hunk 6 above)
Comment (nlyubova) [inline config/fake_controllers.yaml]: done !

File: launch/moveit_rviz.launch (hunk 7 above)
Comment (nlyubova) [inline launch/moveit_rviz.launch]: fixed

File: package.xml (hunk 9 above)
Comment (nlyubova) [inline package.xml:22]: it is auto-generated and yes, I can run demo.launch

**Comment by nlyubova** [comment]
@vrabaud and @keulYSMB thank you for your comments in nao_dcm_robot, nao_virtual and nao_moveit_config. Now, it is much better, could you please accept these three pull requests ? I've tested it with the real Nao and it was grasping! and not shaking
